lib/lists.nix

```
@@ -403,6 +403,45 @@ rec {
   */
   concatMap = builtins.concatMap;
 
+  /**
+    Map, removing null values.
+
+    # Type
+
+    ```
+    filterMap :: (a -> b) -> [a] -> [b]
```

```suggestion
    filterMap :: (a -> (null | b)) -> [a] -> [b]
```

I think we can reasonably convey a bit more info here.

We don't have a real type language, but `|` is for union types in TypeScript and Scala, which is what this is meant to refer to. Other associations with `|` tend to also work.